crate/pyo3/src/py_kms_client.rs

```diff
@@ -24,6 +27,47 @@ use rustls::Certificate;
 
 use crate::py_kms_object::KmsObject;
 
+/// Create a Rekey Keypair request from PyO3 arguments
+/// Returns a PyO3 Future
+macro_rules! rekey_keypair {
+    (
+        $self:ident,
+        $attributes:expr,
+        $master_secret_key_identifier:expr,
+        $tags:expr,
+        $action:expr,
+        $py:ident
+    ) => {{
+        let id = match ($master_secret_key_identifier, $tags) {
+            (Some(key_id), _) => key_id,
+            (_, Some(tags)) => serde_json::to_string(&tags)
+                .map_err(|_e| PyException::new_err("invalid tag(s) specified"))?,
+            _ => return Err(PyException::new_err("please specify a key id or tags")),
+        };
+
+        let policy_attributes = $attributes
+            .into_iter()
+            .map(Attribute::try_from)
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| PyTypeError::new_err(e.to_string()))?;
+
+        let request = build_rekey_keypair_request(&id, policy_attributes, $action)
+            .map_err(|e| PyException::new_err(e.to_string()))?;
+
+        let client = $self.0.clone();
+        pyo3_asyncio::tokio::future_into_py($py, async move {
```

Comment:
You may want to wrap `self.0` into an `Arc` to clone it painlessly. Maybe it is not worth it though, depending on the cost of cloning `self.0` (difficult to judge from this piece of code).

Reply:
We tried to do it once but it implied too many changes in the rust KMS client.